src/current/v22.1/bulk-delete-data.md

```diff
@@ -76,22 +98,185 @@ while True:
 conn.close()
 ~~~
 
-This script iteratively deletes rows in batches of 5,000, until all of the rows where `no_w_id <= 4` are deleted. Note that at each iteration, the filter is updated to match a narrower subset of rows.
+</section>
+
+<section class="filter-content" markdown="1" data-scope="java">
+
+A simple JDBC application that bulk deletes rows in batches of 5000 would look like this:
 
-## Batch delete on a non-indexed column
+{% include_cached copy-clipboard.html %}
+~~~ java
+package com.cockroachlabs.bulkdelete;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.Statement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Optional;
+
+import org.postgresql.ds.PGSimpleDataSource;
+
+public class App {
+
+    public static void deleteData(Connection conn) {
+        boolean cont = true;
+        // the initial warehouse ID we will delete orders from
+        int warehouseId = 4;
+        try {
+            do {
+                System.out.println("Deleting data from warehouses <= to " + warehouseId);
+                String sql = "DELETE FROM new_order WHERE no_w_id <= ? ORDER BY no_w_id DESC LIMIT 5000 RETURNING no_w_id";
+                // use a prepared statement and the current warehouse ID
+                PreparedStatement ps = conn.prepareStatement(sql, ResultSet.TYPE_SCROLL_SENSITIVE, ResultSet.CONCUR_READ_ONLY);
+                ps.setInt(1, warehouseId);
+                ResultSet results = ps.executeQuery();
+                if (!results.next()) {
+                    cont = false;
+                } else {
+                    results.last();
+                    System.out.println("Deleted " + results.getRow() + " rows.");
+                    // get the warehouse ID of the last row of this batch
+                    warehouseId = results.getInt(1);
+                    System.out.println("Warehouse ID is now " + warehouseId);
+                }
+            } while (cont);
+        }
+        catch(Exception e) {
+            return;
+        }
+    }
+
+    public static void main(String[] args) throws SQLException {
+        Connection connection = null;
+        try {
```

Comment:
You should be able to use try-with-resources here to automatically close the connection(please double check this works):
```
    public static void main(String[] args) throws SQLException {
        // create the datasource for the JDBC driver
        PGSimpleDataSource ds = new PGSimpleDataSource();
        ds.setApplicationName("docs_bulk_delete_java");
        // get the cluster JDBC URL from an environment variable
        ds.setUrl(Optional.ofNullable(System.getenv("JDBC_DATABASE_URL")).orElseThrow(
              () -> new IllegalArgumentException("JDBC_DATABASE_URL is not set.")));
        try (Connection connection = ds.getConnection()) {
            // call the method to perform the deletes
            deleteData(connection);
        } catch (SQLException e) {
            e.printStackTrace();
        } 
    }
```

Reply:
Ok, tested it and it works correctly.